File: tests/src/main/java/com/github/klee0kai/test/di/bindinstance/simple_inject/SevenPlanetComponent.java
package com.github.klee0kai.test.di.bindinstance.simple_inject;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.module.BindInstance;
import com.github.klee0kai.stone.closed.IPrivateComponent;
import com.github.klee0kai.test.mowgli.MoonSky;
import com.github.klee0kai.test.mowgli.galaxy.IPlanet;

@Component
public interface SevenPlanetComponent extends IPrivateComponent {


    SevenPlanetModule planets();

    StarsModule stars();

    @BindInstance
    void bindPlanet(IPlanet planet);

    void inject(MoonSky moonSky);
}

File: tests/src/main/java/com/github/klee0kai/test/di/bindinstance/simple_inject/SevenPlanetComponent.kt
package com.github.klee0kai.test.di.bindinstance.simple_inject;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.module.BindInstance;
import com.github.klee0kai.test.mowgli.MoonSky;
import com.github.klee0kai.test.mowgli.galaxy.Earth;
import com.github.klee0kai.test.mowgli.galaxy.IPlanet;
import com.github.klee0kai.test.mowgli.galaxy.Mercury;

@Component
public interface SevenPlanetComponent {


    SevenPlanetModule planets();

    StarsModule stars();

    @BindInstance
    void bind(Earth earth);

    @BindInstance
    void bind(Mercury mercury);

    @BindInstance
    void bindPlanet(IPlanet planet);

    void inject(MoonSky moonSky);
}
